### PR TITLE
Bugfix (Issue 1,2,3)

### PR DESCRIPTION
**Pull request by kelaronn**
Fixed issue #1,#2,#3.

### CODE REVIEW

**Review by fhuszar8 — COMMENTED** [review]
**_Szuper_**

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 1]
```diff
@@ -27,21 +27,21 @@ public TorpedoStore(int numberOfTorpedos){
       }
     }
   }
-
+  Random generator = new Random();
```
Comment (Blasonka) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:30]: fixed

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 2]
```diff
@@ -27,21 +27,21 @@ public TorpedoStore(int numberOfTorpedos){
...
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-      new IllegalArgumentException("numberOfTorpedos");
+      throw new IllegalArgumentException("numberOfTorpedos");
     }
 
     boolean success = false;
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
+
```
Comment (Blasonka) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:39]: fixed

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 3]
```diff
@@ -27,21 +27,21 @@ public TorpedoStore(int numberOfTorpedos){
...
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
+   
     double r = generator.nextDouble();
 
     if (r >= FAILURE_RATE) {
       // successful firing
-      this.torpedoCount =- numberOfTorpedos;
+      this.torpedoCount -= numberOfTorpedos;
```
Comment (Blasonka) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:44]: fixed

**Review by Blasonka — CHANGES_REQUESTED** [review]
comments?

**Pull request merged**